torch/csrc/jit/pickler.cpp

```diff
@@ -551,12 +551,125 @@ void Pickler::pushTuple(const IValue& ivalue) {
   }
 }
 
+static void restoreAccurateTypeTagsIfPossible(const IValue& root) {
```

Comment:
Can we add a comment explaining what this is for? What happens if restoring accurate type tags is *not* possible? The name alone is so tantalizing and full of mystery!